### DISCUSSION TITLE
Understanding the cause of potential variation in care received

### DISCUSSION THREAD

**Discussion by cothurn** [synthetichealth/synthea#1111] [Q&A] [answered]
@jawalonoski 

I am interested in utilizing Synthea for a research project on understanding/quantifying the benefit of preventive checks and procedures. In order to make sure that the generated data is suitable for the research, I am trying to understand what are the causes for variations in care received during the generation process. After inspecting the code and modules,  I have the following questions regarding the generation process.

1. My understanding is that the same module/flowchart is utilized to generate the medical records for all patients. Unless specific characteristics are conditioned upon during transition, every patient utilizes the same probabilities for progressing through the module/flowcharts. For example, all male patients would have the same 10% chance of developing Examplitis, regardless of the patient's previous medical records. Is this understanding correct?
2. While medications and care plans are ordered within modules, I did not find medications or care plans being utilized for transition in the modules that I viewed. My understanding is that medication and care plan orders exist within modules in order to be recorded and have no actual effect. The only two few exceptions I found are not in the .json modules as well—diabetes medicines affect blood sugar levels in [life cycle module](https://github.com/synthetichealth/synthea/blob/d4af94570aba5ba9b199d0071b6ca8be0763e441/src/main/java/org/mitre/synthea/modules/LifecycleModule.java) and blood pressure care clan lowering blood pressure in the [.java file generating blood pressure values](https://github.com/synthetichealth/synthea/blob/7e4eca3219925a22e8e1e0dfbb50a824b0cfa517/src/main/java/org/mitre/synthea/modules/BloodPressureValueGenerator.java). Is it correct to assume that, at large, medications and care plans have no impact on patient disease progression in the generated data?
3. Based on [encounterModule.java](https://github.com/synthetichealth/synthea/blob/d4af94570aba5ba9b199d0071b6ca8be0763e441/src/main/java/org/mitre/synthea/modules/EncounterModule.java), the frequency of wellness encounters is predetermined given a patient's age. Would it be appropriate to introduce some simple variation by editing [this function](ba5ba9b199d0071b6ca8be0763e441/src/main/java/org/mitre/synthea/modules/EncounterModule.java#L180)? 
4. Besides the wellness encounters generated based on pre-determined intervals and encounters specified in the modules, is there any other way encounters can be generated? For example, the metabolic syndrome care/disease modules do not include ED visits due to diabetes complications. Is it correct to assume that, because of this, there would be no ED visits due to diabetes complications generated, regardless of the parameters used?

Sorry for the long-winded questions, and thanks in advance.

TJ

**Comment by jawalonoski** [comment, marked as answer]
1. Correct. Some modules take into account existing medical conditions, and some do not. It depends on the author and the level of complexity that was being targeted.
2. There are only 3 instances of `"condition_type": "Active CarePlan"` transition logic, and 56 instances of `"condition_type": "Active Medication"` transition logic today. Ideally, those transitions should be based on medication class (e.g. ACE inhibitor or NSAIDs) rather than individual drugs, but we don't model medication class today. Unless modeled explicitly, patients are assumed to adhere to the medications they are prescribed. In many cases, the impacts of these medications on the patient are "built-in" to the module, and many do not alter lab values or vital signs. Some do though. Finally, Synthea does not have a drug-drug interaction model, so those impacts are not taken into account.
3. The link you provided to "this function" was broken, so I can't see it. The answer is "maybe" it is that function, and "maybe" it is appropriate. It depends on what exactly you are trying to do.
4. In Java, the `CardiovascularDiseaseModule`, the `DeathModule`, the `C19ImmunizationModule`, and the `EncounterModule` can all generate encounters outside of the JSON modules. Creating `Symptom` states whose summed values that collectively exceed thresholds defined in the `EncounterModule` can trigger unplanned emergency, urgent, or PCP encounters. The diabetes modules do not use these, to my knowledge.

**Reply by cothurn** [reply]
Thank you for the quick response, and I really appreciate the detailed explanation.

A bit of follow-up with regard to Q3. Here is a working link to the function that I was referring to —[recommendedTimeBetweenWellnessVisits](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/EncounterModule.java#L180). The generated value is being utilized here on [line 70](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/EncounterModule.java#L70). I would like to introduce some variation to the time between two wellness encounters—for example, allowing people of age between 20 and 39 to have a wellness encounter every 1~3 years randomly instead of a fixed value of every 3 years. Would this be the place to make some edits?  

Thanks again.

**Reply by jawalonoski** [reply]
The answer to Q3 is "yes."

However, introducing variation that way may not actually alter outcomes that dramatically. If you wanted the variation to actually change disease progression you may have to make additional changes in other areas.

**Reply by cothurn** [reply]
Thank you for the response.

**Comment by jawalonoski** [comment]
I would also add that we have done experiments altering the modules to have different pathways based on preventative medicine and interventions (for example, from a Community Health Worker). We measured longitudinal impacts based on QALYs, DALYs, and lifetime costs. That code isn't merged into the `master` branch, but you can do what-if experiments with Synthea comparing different synthetic cohorts if you are willing to alter the modules and code according to the research and literature and perform some level of validation and verification commensurate with your use-case.
